### PR TITLE
Added wdio E2E capabilities

### PR DESCRIPTION
**Pull request by stefancplace**

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -94,6 +94,14 @@ const cli = meow(`
             --test-sources
                 Will refactor an old plugin source structure like 'src/classes' or 'src/java' to a proper Maven-like structure with
                 'src/main/java' and 'src/test/java'
+        
+        e2e --baseUrl <baseUrl> [--browser <browser>] [--plugins <plugins>]
```
Comment (rose-m) [inline src/cli.ts]: If `--baseUrl` is an optional parameter then we should write `[--baseUrl <baseUrl>]`

File: src/cli.ts [hunk 2]
```diff
@@ -94,6 +94,14 @@ const cli = meow(`
...
+        
+            --baseUrl to configure where to run the tests against (default: localhost)
+            --browser to specify which browser to use (default: Chrome?)
```
Comment (rose-m) [inline src/cli.ts]: `default: Chrome` (no `?`)?

File: src/commands/e2e/E2E.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            this.baseUrl = baseUrl;
+        } else {
+            this.baseUrl = E2E.DEFAULT_BASE_URL;
+        }
+
+        const browser = params[E2E.BROWSER];
+        if (typeof browser === 'string' && browser.length > 0) {
+            if (browser === 'ie11' || browser.startsWith('internet')) {
+                this.browser = E2E.INTERNET_EXPLORER;
+                this.browserVersion = '11';
```
Comment (rose-m) [inline src/commands/e2e/E2E.ts]: Never used - remove

File: src/commands/e2e/ConfigTemplate.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import {E2E} from './E2E';
+
+export class ConfigTemplate {
+    private readonly template: string;
+
+    constructor(e2eFolder: string, browser: string, baseUrl: string, timeout: number) {
```
Comment (rose-m) [inline src/commands/e2e/ConfigTemplate.ts]: We have to test this under Windows to make sure the generated paths are used correctly 👍

File: src/commands/e2e/E2E.ts (hunk 3 above)
Comment (stefancplace) [reply src/commands/e2e/E2E.ts]: removed it, in bash you have to write --browser="internet explorer" then, might be a bit confusing, probably to be added in the --help ?

File: src/commands/e2e/ConfigTemplate.ts (hunk 4 above)
Comment (stefancplace) [reply src/commands/e2e/ConfigTemplate.ts]: I tested it with Audrius on Windows and fixed e2eFolder, should be fine now

File: src/commands/e2e/E2E.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            this.timeout = timeout;
+        } else {
+            this.timeout = E2E.DEFAULT_TIMEOUT;
+        }
+
+        const headless = params[E2E.HEADLESS];
+        if (typeof headless === 'boolean') {
+            this.headless = true;
+        } else {
+            this.headless = E2E.DEFAULT_HEADLESS;
```
Comment (rose-m) [inline src/commands/e2e/E2E.ts]: Check the combination of headless and browser - atm the comment says it's only possible for Chrome.

File: src/commands/e2e/E2E.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        const timeout = params[E2E.TIMEOUT];
+        if (typeof timeout === 'number') {
+            this.timeout = timeout;
+        } else {
+            this.timeout = E2E.DEFAULT_TIMEOUT;
+        }
+
+        const headless = params[E2E.HEADLESS];
+        if (typeof headless === 'boolean') {
+            this.headless = true;
```
Comment (rose-m) [inline src/commands/e2e/E2E.ts]: `this.headless = headless`;

File: src/commands/e2e/E2E.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+import {Global} from '../../Global';
+import * as path from 'path';
+import * as fs from 'fs';
+
+export class E2E implements ICommand {
+    private static readonly BASE_URL: string = 'baseUrl';
+    private static readonly PLUGINS: string = 'plugins';
+    private static readonly BROWSER: string = 'browser';
+
+    private static readonly TIMEOUT: string = 'timeout';
```
Comment (rose-m) [inline src/commands/e2e/E2E.ts]: Best prefix all CLI parameters with `PARAMETER_` for consistency :)

File: src/commands/e2e/E2E.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        } else {
+            this.headless = E2E.DEFAULT_HEADLESS;
+        }
+
+        return true;
+    }
+
+    public execute(): Promise<void> {
+        const wdioGenerator = new WdioConfigGenerator(this.pluginsToBeTested, this.baseUrl, this.browser, this.timeout, this.workingDir, this.headless);
+        console.log('Generation WDIO configuration files...');
```
Comment (rose-m) [inline src/commands/e2e/E2E.ts]: Typo: `Generating`

**Pull request merged**